packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,604 @@
+import * as React from 'react';
+import {
+  initializeComponentRef,
+  classNamesFunction,
+  KeyCodes,
+  getRTLSafeKeyCode,
+  css,
+  getId,
+  EventGroup,
+  IRenderFunction,
+} from '../../Utilities';
+import { List, ScrollToMode, IListProps } from '../../List';
+import { ISelection, SelectionMode, SELECTION_CHANGE } from '../../Selection';
+import { FocusZone, FocusZoneDirection } from '../../FocusZone';
+import type { IProcessedStyleSet } from '../../Styling';
+import type {
+  IGroupedList,
+  IGroupedListProps,
+  IGroup,
+  IGroupRenderProps,
+  IGroupedListStyleProps,
+  IGroupedListStyles,
+} from './GroupedList.types';
+import { GroupHeader } from './GroupHeader';
+import { GroupShowAll } from './GroupShowAll';
+import { GroupFooter } from './GroupFooter';
+import type { IGroupHeaderProps } from './GroupHeader';
+import type { IGroupShowAllProps } from './GroupShowAll.styles';
+import type { IGroupFooterProps } from './GroupFooter.types';
+import { useMount, useUnmount } from '@fluentui/react-hooks';
+
+export interface IGroupedListV2State {
+  selectionMode?: IGroupedListProps['selectionMode'];
+  compact?: IGroupedListProps['compact'];
+  groups?: IGroup[];
+  items?: IGroupedListProps['items'];
+  listProps?: IGroupedListProps['listProps'];
+  version: {};
+  groupExpandedVersion: {};
+}
+
+export interface IGroupedListV2Props extends IGroupedListProps {
+  listRef: React.Ref<List>;
+  version: {};
+  groupExpandedVersion: {};
+}
+
+type IITemGroupedItem = {
+  type: 'item';
+  group: IGroup;
+  item: any;
+  itemIndex: number;
+};
+
+type IShowAllGroupedItem = {
+  type: 'showAll';
+  group: IGroup;
+};
+
+type IFooterGroupedItem = {
+  type: 'footer';
+  group: IGroup;
+};
+
+type IHeaderGroupedItem = {
+  type: 'header';
+  group: IGroup;
+  groupId: string;
+};
+
+type IGroupedItem = IITemGroupedItem | IShowAllGroupedItem | IFooterGroupedItem | IHeaderGroupedItem;
+
+type FlattenItemsFn = (
```

Comment:
Nit:  The Fn suffix feels hungarian to me.

Reply:
Dropped!